Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,148 @@ static void Main(string[] args)
+            String FirstNumberString="", SecondNumberString="", MultiplicationResultString="";
+            int MultiplicationPosition = Equation.IndexOf("*");
+            int equalsPosition = Equation.IndexOf("=");
+            
+            MultiplicationResultString = Equation.Substring(equalsPosition+1);
+            SecondNumberString = Equation.Substring(MultiplicationPosition+1, equalsPosition-MultiplicationPosition-1);
+            FirstNumberString = Equation.Substring(0, MultiplicationPosition);
+
+            //Converting FirstNumberString SecondNumberString MultiplicationResultString into no.
+            if(FirstNumberString.IndexOf('?')<0){
+                FirstNumber = Int32.Parse(FirstNumberString);
+            }
+            if(SecondNumberString.IndexOf('?')<0){
+                SecondNumber = Int32.Parse(SecondNumberString);
+            }
+            if(MultiplicationResultString.IndexOf('?')<0){
+                MultiplicationResult = Int32.Parse(MultiplicationResultString);
+            }
+
+            //Checking for missing digit
+            if(MultiplicationResultString.IndexOf('?')>=0){
+                MissingNumber = missingDigitInResult(FirstNumber,SecondNumber,MultiplicationResultString);
+            }
+
+            if(FirstNumberString.IndexOf('?')>=0){
+                MissingNumber = missingDigitInFirstNumber(MultiplicationResult, SecondNumber, FirstNumberString);
+            }
+
+            if(SecondNumberString.IndexOf('?')>=0){
+                MissingNumber = missingDigitInSecondNumber(MultiplicationResult, FirstNumber, SecondNumberString);
+            }
+
+
+            return MissingNumber;
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int missingDigitInResult(int FirstNumber, int SecondNumber, string MultiplicationResultString){
+            int MissingNumber = -1;
+            int answer = FirstNumber*SecondNumber;
+                String answerString = answer.ToString();
+                int position = MultiplicationResultString.IndexOf('?');
+                char result = answerString[position];
+                MissingNumber = result - '0';
+                String verifyAnswer = MultiplicationResultString.Replace('?', result);
+
+                if(MultiplicationResultString.IndexOf('?')==0){
+                    if(answerString.Equals(MultiplicationResultString.Substring(1))){
+                        MissingNumber=-1;
+                    }
+                }
+                if(answerString.Equals(verifyAnswer)){
+                    MissingNumber = result - '0';
+                }else{
+                    MissingNumber = -1;
+                }
+
+                return MissingNumber;
+        }
+
+        public static int missingDigitInFirstNumber(int MultiplicationResult, int SecondNumber, string FirstNumberString){

```

Comment:
can you get rid of `missingDigitInSecondNumber` and use `missingDigitInFirstNumber` for both.
Also better name would be `GetMissingDigitInOperand(int result, int operand1,string operandWithMissingDigit)`

Reply:
Casing standards are different in different languages. For c# Method names are Pascal, while local variables and method parameters are Camel.
See methods like `IndexOf` , `ToString` you can see parameter names when you hover/go to defination on these.
Refer https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions